packages/internals/loader/src/loader.test.js

```diff
@@ -0,0 +1,74 @@
+const {join} = require('path');
+const Resolver = require('enhanced-resolve/lib/Resolver');
+const {ResolverFactory} = require('enhanced-resolve');
+const {expect, sinon} = require('../../../../test/helpers');
+const Loader = require('./loader');
+
+describe('Loader', function () {
+  describe('constructor()', function () {
+    it('passes configuration options to the resolver', function () {
+      const spy = sinon.spy(ResolverFactory, 'createResolver');
+      // eslint-disable-next-line no-unused-vars
+      const loader = new Loader({
+        fileSystem: 'foo'
+      });
+      expect(spy.args[0][0].fileSystem).to.equal('foo');
+      spy.restore();
+    });
+  });
+
+  describe('.resolve()', function () {
+    it('synchronously resolves the path with respect to the root', function () {
+      const loader = new Loader();
+      expect(loader.resolve('./loader', __dirname)).to.equal(join(__dirname, 'loader.js'));
+    });
+
+    it('resolved paths relative to the cwd if no root is supplied', function () {
+      const loader = new Loader();
+      const result = loader.resolve('./test/fixtures/add-ons/plugin');
+      expect(result).to.equal(join(process.cwd(), 'test/fixtures/add-ons/plugin.js'));
+    });
+
+    it('throws an error if the path cannot be resolved', function () {
+      const loader = new Loader();
+      expect(() => loader.resolve('./foo', __dirname)).to.throw();
```

Comment:
Worth adding/wrapping our own error message here? I'd say this could be a fairly common error for people to encounter.